background/services/enrichment/transactions.ts

```diff
@@ -244,7 +244,16 @@ export default async function resolveTransactionAnnotation(
   // If the wallet doesn't have enough base asset to cover gas, push a warning
   if (gasFee + (transaction.value ?? 0n) > baseAssetBalance) {
     txAnnotation.warnings ??= []
-    txAnnotation.warnings.push("insufficient-funds")
+    if (!txAnnotation.warnings.includes("insufficient-funds")) {
+      txAnnotation.warnings.push("insufficient-funds")
+    }
+  } else if (txAnnotation.warnings) {
```

Comment:
We can move the `txAnnotation.warnings ??= []` outside the parent conditional block and change this to
```suggestion
  } else {
```
This way we can treat this handler as "either we have enough to cover gas costs or not" instead of "if we don't have enough for gas but have warnings..."